js/generate.js

```diff
@@ -0,0 +1,51 @@
+// get the device size before get the image
+
+let width = "1440";
+let height = "2960";
+
+if (window.screen.width >= 1024) {
+  width = "1920";
+  height = "1080";
```

Comment:
Make the width static and give the height a dynamic value for all your cases
It can be `height = width * window.screen.width/window.screen.height`